app/src/main/java/id/rizmaulana/covid19/ui/overview/DashboardActivity.kt

```diff
@@ -1,30 +1,41 @@
 package id.rizmaulana.covid19.ui.overview
 
-import android.animation.ValueAnimator
-import android.graphics.Color
 import android.os.Bundle
-import android.widget.TextView
-import com.github.mikephil.charting.animation.Easing
-import com.github.mikephil.charting.data.PieData
-import com.github.mikephil.charting.data.PieDataSet
-import com.github.mikephil.charting.data.PieEntry
+import android.util.Log
+import android.view.View
 import id.rizmaulana.covid19.R
-import id.rizmaulana.covid19.data.model.CovidDaily
-import id.rizmaulana.covid19.data.model.CovidOverview
 import id.rizmaulana.covid19.databinding.ActivityDashboardBinding
 import id.rizmaulana.covid19.ui.adapter.DailyAdapter
+import id.rizmaulana.covid19.ui.adapter.DailyAdapterItemClickListener
+import id.rizmaulana.covid19.ui.adapter.viewholders.DailyItem
+import id.rizmaulana.covid19.ui.adapter.viewholders.OverviewItem
 import id.rizmaulana.covid19.ui.base.BaseActivity
+import id.rizmaulana.covid19.ui.base.BaseViewItem
 import id.rizmaulana.covid19.ui.detail.DetailActivity
 import id.rizmaulana.covid19.util.CaseType
-import id.rizmaulana.covid19.util.NumberUtils
-import id.rizmaulana.covid19.util.ext.color
 import id.rizmaulana.covid19.util.ext.observe
 import org.koin.android.viewmodel.ext.android.viewModel
 
 class DashboardActivity : BaseActivity() {
 
     private val viewModel by viewModel<DashboardViewModel>()
-    private val dailyAdapter by lazy { DailyAdapter() }
+    private val dailyAdapter by lazy { DailyAdapter(object: DailyAdapterItemClickListener {
+            override fun invoke(viewItem: BaseViewItem, view: View) {
+               when(viewItem) {
+                   is OverviewItem -> {
+                       when(view.id) {
+                           R.id.layout_confirmed -> permission(CaseType.CONFIRMED)
+                           R.id.layout_recovered -> permission(CaseType.RECOVERED)
+                           R.id.layout_death -> permission(CaseType.DEATHS)
+                       }
+                   }
+                   is DailyItem -> {
+                      Log.e("DailyItem", "DailyItem Click: ${viewItem.deltaConfirmed}")
+                   }
+               }
+            }
+        })
```

Comment:
to simplify the instance of `dailyAdapter`, maybe you can separate this one with a new callback function. i.e.:

```kt
...
private val dailyAdapter by lazy { DailyAdapter(::onItemClicked) }
...

private fun onItemClicked(viewItem: BaseViewItem, view: View) {
    //do something
}
```

Reply:
Thank you! I dont know it can be like that 😆